Student/Student/Submissions/SubmissionComments/SubmissionCommentsViewController.swift

```diff
@@ -162,38 +166,39 @@ extension SubmissionCommentsViewController: AudioRecorderDelegate {
     }
 }
 
+extension SubmissionCommentsViewController: UIImagePickerControllerDelegate, UINavigationControllerDelegate {
+    func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [UIImagePickerController.InfoKey: Any]) {
+        guard let url = info[.mediaURL] as? URL else { return }
+        presenter?.addMediaComment(type: .video, url: url)
+        picker.dismiss(animated: true)
+    }
+
+    func imagePickerControllerDidCancel(_ picker: UIImagePickerController) {
+        picker.dismiss(animated: true)
+    }
+}
+
 extension SubmissionCommentsViewController: SubmissionCommentsViewProtocol {
     func reload() {
         emptyLabel?.isHidden = presenter?.comments.isEmpty == false
         guard let changes = presenter?.comments.changes, changes.count == 1 else {
             tableView?.reloadData()
             return
         }
-        tableView?.performBatchUpdates({
-            for change in changes {
-                switch change {
-                case .insertSection(let section):
-                    tableView?.insertSections([section], with: .automatic)
-                case .deleteSection(let section):
-                    tableView?.deleteSections([section], with: .automatic)
-                case .insertRow(let row):
-                    tableView?.insertRows(at: [
-                        IndexPath(row: row.row * 2, section: row.section),
-                        IndexPath(row: row.row * 2 + 1, section: row.section),
-                    ], with: .automatic)
-                case .updateRow(let row):
-                    tableView?.reloadRows(at: [
-                        IndexPath(row: row.row * 2, section: row.section),
-                        IndexPath(row: row.row * 2 + 1, section: row.section),
-                    ], with: .automatic)
-                case .deleteRow(let row):
-                    tableView?.deleteRows(at: [
-                        IndexPath(row: row.row * 2, section: row.section),
-                        IndexPath(row: row.row * 2 + 1, section: row.section),
-                    ], with: .automatic)
-                }
-            }
-        })
+        switch changes[0] {
+        case .insertRow(let row):
+            tableView?.insertRows(at: [
+                IndexPath(row: row.row * 2, section: row.section),
+                IndexPath(row: row.row * 2 + 1, section: row.section),
+            ], with: .automatic)
+        case .updateRow(let row):
+            tableView?.reloadRows(at: [
+                IndexPath(row: row.row * 2, section: row.section),
+                IndexPath(row: row.row * 2 + 1, section: row.section),
+            ], with: .automatic)
+        default:
+            tableView?.reloadData()
+        }
```

Comment:
since we only know we want the animation for a single added placeholder row or the replaced final row, bail out to reloadData in more cases.